crates/ty/src/lib.rs

```diff
@@ -143,7 +144,9 @@ fn run_check(args: CheckCommand) -> anyhow::Result<ExitStatus> {
         main_loop.run(&mut db)?
     };
 
-    tracing::trace!("Counts for entire CLI run:\n{}", countme::get_all());
+    if std::env::var("TY_MEMORY_REPORT").as_deref() == Ok("1") {
+        salsa_memory_dump(&db);
+    }
```

Comment:
It would be nice if:

* We would print a condenced memory report when running with `-vv` 
* We would print the full memory report when running with `-vvv`

You can get the verbosity from `args.verbosity`. 

I would probably skip the environment variable for now. If we don't, make sure to add it here https://github.com/astral-sh/ty/blob/main/docs/reference/env.md